merkle/merkle_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package merkle
+
+import (
+	"context"
+	"crypto/rand"
+	"strconv"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	resRoot ids.ID
+	resDB   merkledb.MerkleDB
+	resErr  error
+)
+
+func BenchmarkMerkleTxRoot(b *testing.B) {
+	for _, size := range []int{10, 100, 1000, 10000} {
+		ctx := context.TODO()
+		tracer := trace.Noop
+		merkleItems := make([][]byte, 0, size)
+		for i := 0; i < size; i++ {
+			item := make([]byte, 32)
+			_, err := rand.Read(item)
+			require.NoError(b, err)
+			merkleItems = append(merkleItems, item)
+		}
+
+		var root ids.ID
+		var db merkledb.MerkleDB
+		var err error
+
+		defaultConfig := merkledb.Config{
```

Comment:
I don't really want to merge this until this config is largely removed. We need a "single instance" trie that is merkleDB-compatible for #950